impl/c-qube/src/services/dimension/dimension.service.spec.ts

```diff
@@ -3,99 +3,183 @@ import { DimensionGrammar } from 'src/types/dimension';
+    jest.spyOn(qbService, 'generateIndexStatement').mockReturnValue(indexQuery);
+
+    const $queryRawUnsafeSpy = jest.spyOn(prismaService, '$queryRawUnsafe').mockRejectedValue(error);
+    jest.spyOn(console, 'error').mockImplementation();
+    await service.createDimension(mockDimensionGrammar(), autoPrimaryKey);
+
+    expect(qbService.generateCreateStatement).toHaveBeenCalledWith(mockDimensionGrammar().schema, autoPrimaryKey);
+    expect(qbService.generateIndexStatement).toHaveBeenCalledWith(mockDimensionGrammar().schema);
+    expect(prismaService.$queryRawUnsafe).toHaveBeenCalledWith(createQuery);
+    expect(console.error).toHaveBeenCalledWith(mockDimensionGrammar().name);
+    expect(console.error).toHaveBeenCalledWith(JSON.stringify(mockDimensionGrammar(), null, 2));
+    expect(console.error).toHaveBeenCalledWith({ createQuery });
+    expect(console.error).toHaveBeenCalledWith({ indexQuery });
+  });
+
+  it('should insert bulk dimension data and handle errors', async () => {
+    const mockData = [
+       { name: 'Item 1', id: 1 },
+       { name: 'Item 2', id: 2 },
+    ];
+    const mockInsertQuery = 'mock-insert-query';
+    const mockError = new Error('Some error occurred');
+
+    jest.spyOn(qbService, 'generateBulkInsertStatementOld').mockReturnValue(mockInsertQuery);
+    prismaService.$queryRawUnsafe = jest.fn().mockRejectedValue(mockError);
+    console.error = jest.fn(); 
+    console.log = jest.fn(); 
+
+    await service.insertBulkDimensionDataV2(mockDimensionGrammar(), mockData);
+
+    expect(qbService.generateBulkInsertStatementOld).toHaveBeenCalledWith(mockDimensionGrammar().schema, mockData);
+    expect(prismaService.$queryRawUnsafe).toHaveBeenCalledWith(mockInsertQuery);
+    expect(console.error).toHaveBeenCalledWith(mockInsertQuery);
+    if (mockData.length < 50) {
+      expect(console.log).toHaveBeenCalledWith(mockData);
+    }
+    expect(console.error).toHaveBeenCalledWith(mockDimensionGrammar().name);
+    expect(console.error).toHaveBeenCalledWith(mockError);
+  });
+
+  it('should insert bulk dimension data and handle errors', async () => {
+    const mockData = [
+    ];
+    const mockInsertQuery = 'mock-insert-query';
+    const mockError = new Error('Some error occurred');
+
+    jest.spyOn(qbService, 'generateBulkInsertStatementOld').mockReturnValue(mockInsertQuery);
+    prismaService.$queryRawUnsafe = jest.fn().mockRejectedValue(mockError);
+    console.error = jest.fn();
+    console.log = jest.fn(); 
+    await service.insertBulkDimensionData(mockDimensionGrammar(), mockData);
+
+    expect(qbService.generateBulkInsertStatementOld).toHaveBeenCalledWith(mockDimensionGrammar().schema, mockData);
+    expect(prismaService.$queryRawUnsafe).toHaveBeenCalledWith(mockInsertQuery);
+    if (mockData.length < 50) {
+      expect(console.log).toHaveBeenCalledWith(mockData);
+    }
+    expect(console.error).toHaveBeenCalledWith(mockDimensionGrammar().name);
```

Comment:
why are we checking console logging statements?

Reply:
Having logs on the console during tests was not a problem, anchoring entire tests on consoles is a big problem